common/utils/src/main/resources/error/error-classes.json

```diff
@@ -1823,6 +1836,18 @@
     },
     "sqlState" : "HY109"
   },
+  "INVALID_DATETIME_PATTERN" : {
+    "message" : [
+      "Unrecognized datetime pattern: <pattern>. Please provide correct datetime pattern."
+    ],
+    "sqlState" : "22007"
+  },
+  "INVALID_DATETIME_PATTERN_LENGTH" : {
```

Comment:
Could you create a sub-class of `INVALID_DATETIME_PATTERN` with the name `LENGTH`

Reply:
Done in new commits. Changed relevant tests also.